### ISSUE TITLE
VERI-Wild issue

### ISSUE THREAD

**Issue by levipereira** [JDAI-CV/fast-reid#488]
I'm trying to use [pre-trained model VERI-Wild](https://github.com/JDAI-CV/fast-reid/blob/master/MODEL_ZOO.md#vehicleid-baseline) on [FastMot](https://github.com/GeekAlexis/FastMOT).

I exported the VERI-Wild model to onnx and it has a feature dimension of 2048.  
Also, I checked the ONNX model using np.testing.assert_allclose(torch_out, ort_out, rtol=1e-3, atol=1e-6) and the output of the original pytorch model doesn't match the onnx version.
https://github.com/GeekAlexis/FastMOT/issues/86#issuecomment-826424817

I should to re-train the VERI-Wild model or I can use the pre-trained model?

When I tried to train VERI-Wild dataset the error below is raised:
```
[05/14 16:05:08 fastreid.engine.defaults]: Prepare training set
Traceback (most recent call last):
  File "tools/train_net.py", line 57, in <module>
    args=(args,),
  File "./fastreid/engine/launch.py", line 71, in launch
    main_func(*args)
  File "tools/train_net.py", line 42, in main
    trainer = DefaultTrainer(cfg)
  File "./fastreid/engine/defaults.py", line 205, in __init__
    data_loader = self.build_train_loader(cfg)
  File "./fastreid/engine/defaults.py", line 416, in build_train_loader
    return build_reid_train_loader(cfg, combineall=cfg.DATASETS.COMBINEALL)
  File "./fastreid/config/config.py", line 264, in wrapped
    explicit_args = _get_args_from_config(from_config, *args, **kwargs)
  File "./fastreid/config/config.py", line 292, in _get_args_from_config
    ret = from_config_func(*args, **kwargs)
  File "./fastreid/data/build.py", line 36, in _train_loader_from_config
    data = DATASET_REGISTRY.get(d)(root=_root, **kwargs)
  File "./fastreid/data/datasets/veriwild.py", line 53, in __init__
    train = self.process_dir(self.train_list)
  File "./fastreid/data/datasets/veriwild.py", line 69, in process_dir
    dataset.append((self.imgid2imgpath[imgid], vid, self.imgid2camid[imgid]))
KeyError: '119083.jpg 0 23'
```

**Comment by viyer-research** [comment]
As you have successfully exported to fastreid->ONNX, could you please share the path setup for the ONNX utility as I am getting an error if my files VeRi files are in ~/Download directory. I am using the e.g. below but may root-path is --config-file ~/Download/sbs_R50-ibn.yml --name baselineR50 --output outputs/onnx_model ~/Downloads/veri_sbs_R50-ibn.pth


python onnx_export.py --config-file root-path/bagtricks_R50/config.yml --name baseline_R50 --output outputs/onnx_model --opts MODEL.WEIGHTS root-path/logs/market1501/bagtricks_R50/model_final.pth

Thanks,

**Comment by JinkaiZheng** [comment]
@levipereira Hi~ We found the reason. The latest official VERI-Wild TXT files have updated on 2020.09.07, however, we used the 2019 TXT version when we developed this system. 
That's not a big deal, just change one line of the code (please refer to [this](https://github.com/JDAI-CV/fast-reid/pull/507/commits/36bcd969a6624d14c70541ba8a655214b2488d57)), and you'll be able to match the latest VERI-Wild TXT version.

**Comment by levipereira** [comment]
@JinkaiZheng  this change worked except to one line in txt file raise error during 
```
000001 00001/000001.jpg 14403 14 1
```


```
[06/16 16:09:49 fastreid]: Full config saved to /mnt/docker/repos_git/fast-reid/logs/veriwild/bagtricks_R50-ibn_4gpu/config.yaml
[06/16 16:09:49 fastreid.utils.env]: Using a generated random seed 49486628
[06/16 16:09:49 fastreid.engine.defaults]: Prepare training set

Traceback (most recent call last):
  File "tools/train_net.py", line 57, in <module>
    args=(args,),
  File "./fastreid/engine/launch.py", line 71, in launch
    main_func(*args)
  File "tools/train_net.py", line 42, in main
    trainer = DefaultTrainer(cfg)
  File "./fastreid/engine/defaults.py", line 201, in __init__
    data_loader = self.build_train_loader(cfg)
  File "./fastreid/engine/defaults.py", line 400, in build_train_loader
    return build_reid_train_loader(cfg, combineall=cfg.DATASETS.COMBINEALL)
  File "./fastreid/config/config.py", line 264, in wrapped
    explicit_args = _get_args_from_config(from_config, *args, **kwargs)
  File "./fastreid/config/config.py", line 292, in _get_args_from_config
    ret = from_config_func(*args, **kwargs)
  File "./fastreid/data/build.py", line 36, in _train_loader_from_config
    data = DATASET_REGISTRY.get(d)(root=_root, **kwargs)
  File "./fastreid/data/datasets/veriwild.py", line 53, in __init__
    train = self.process_dir(self.train_list)
  File "./fastreid/data/datasets/veriwild.py", line 71, in process_dir
    print(self.imgid2imgpath[imgid])
KeyError: '000001'
```
As Workaround I removed this line from train_test.txt and worked.


But this new error is raised during training... how to fix it?
```
[06/16 06:33:57 fastreid.utils.events]:  eta: 1 day, 21:06:04  epoch/iter: 9/43399  total_loss: 5.984  loss_cls: 5.923  loss_triplet: 0.04957  time: 0.3407  data_time: 0.0018  lr: 3.50e-04  max_mem: 6332M
[06/16 06:33:57 fastreid.engine.defaults]: Prepare testing set
[06/16 06:33:59 fastreid.data.datasets.bases]: => Loaded SmallVeRiWild in csv format: 
| subset   | # ids   | # images   | # cameras   |
|:---------|:--------|:-----------|:------------|
| query    | 3000    | 3000       | 105         |
| gallery  | 3000    | 38861      | 146         |
[06/16 06:33:59 fastreid.evaluation.evaluator]: Start inference on 41861 images
ERROR [06/16 06:34:01 fastreid.engine.train_loop]: Exception during training:
Traceback (most recent call last):
  File "./fastreid/engine/train_loop.py", line 147, in train
    self.after_epoch()
  File "./fastreid/engine/train_loop.py", line 181, in after_epoch
    h.after_epoch()
  File "./fastreid/engine/hooks.py", line 377, in after_epoch
    self._do_eval()
  File "./fastreid/engine/hooks.py", line 351, in _do_eval
    results = self._func()
  File "./fastreid/engine/defaults.py", line 303, in test_and_save_results
    self._last_eval_results = self.test(self.cfg, self.model)
  File "./fastreid/engine/defaults.py", line 439, in test
    results_i = inference_on_dataset(model, data_loader, evaluator, flip_test=cfg.TEST.FLIP.ENABLED)
  File "./fastreid/evaluation/evaluator.py", line 126, in inference_on_dataset
    evaluator.process(inputs, outputs)
  File "./fastreid/evaluation/reid_evaluation.py", line 43, in process
    'camids': inputs['camids'].to(self._cpu_device)
AttributeError: 'list' object has no attribute 'to'
[06/16 06:34:01 fastreid.engine.hooks]: Overall training speed: 43398 iterations in 4:06:27 (0.3407 s / it)
[06/16 06:34:01 fastreid.engine.hooks]: Total training time: 4:07:54 (0:01:26 on hooks)
Traceback (most recent call last):
  File "tools/train_net.py", line 57, in <module>
    args=(args,),
  File "./fastreid/engine/launch.py", line 71, in launch
    main_func(*args)
  File "tools/train_net.py", line 45, in main
    return trainer.train()
  File "./fastreid/engine/defaults.py", line 348, in train
    super().train(self.start_epoch, self.max_epoch, self.iters_per_epoch)
  File "./fastreid/engine/train_loop.py", line 147, in train
    self.after_epoch()
  File "./fastreid/engine/train_loop.py", line 181, in after_epoch
    h.after_epoch()
  File "./fastreid/engine/hooks.py", line 377, in after_epoch
    self._do_eval()
  File "./fastreid/engine/hooks.py", line 351, in _do_eval
    results = self._func()
  File "./fastreid/engine/defaults.py", line 303, in test_and_save_results
    self._last_eval_results = self.test(self.cfg, self.model)
  File "./fastreid/engine/defaults.py", line 439, in test
    results_i = inference_on_dataset(model, data_loader, evaluator, flip_test=cfg.TEST.FLIP.ENABLED)
  File "./fastreid/evaluation/evaluator.py", line 126, in inference_on_dataset
    evaluator.process(inputs, outputs)
  File "./fastreid/evaluation/reid_evaluation.py", line 43, in process
    'camids': inputs['camids'].to(self._cpu_device)
AttributeError: 'list' object has no attribute 'to'
```

**Comment by Soar-Sir** [comment]
I met the same problem.

**Comment by L1aoXingyu** [comment]
fix it in d792a69f3f69159ba3ca6341a8be0e6da2b31633
